update.check/src/main/java/org/kaazing/gateway/update/check/UpdateCheckTask.java

```diff
@@ -0,0 +1,105 @@
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An update check task that updates the latest version
+ */
+public class UpdateCheckTask implements Runnable {
+
+    private final Logger logger = LoggerFactory.getLogger(UpdateCheckTask.class);
+    private final UpdateCheckGatewayObserver updateCheckGatewayObserver;
+    private final String protocolVersion = "1.0";
+    private final String versionServiceUrl;
+
+    /**
+     * @param updateCheckGatewayObserver
+     * @param webServiceUrl
+     * @param productName
+     */
+    public UpdateCheckTask(UpdateCheckGatewayObserver updateCheckGatewayObserver, String webServiceUrl, String productName) {
+        this.updateCheckGatewayObserver = updateCheckGatewayObserver;
+        if (webServiceUrl.endsWith("/")) {
+            this.versionServiceUrl = format("%s%s/%s/latest", webServiceUrl, productName, protocolVersion);
+        } else {
+            this.versionServiceUrl = format("%s/%s/%s/latest", webServiceUrl, productName, protocolVersion);
+        }
+    }
+
+    @Override
+    public void run() {
+        GatewayVersion latestVersion = fetchLatestVersion();
+        if (latestVersion != null) {
+            updateCheckGatewayObserver.setLatestGatewayVersion(latestVersion);
+        }
+    }
+
+    /**
+     * Checks Kaazing Version Web Service to see what is the latest version of the product
+     *
+     * @return null if an exception is caught, otherwise returns the latest version of the gateway
+     */
+    private GatewayVersion fetchLatestVersion() {
+        GatewayVersion latestVersion = null;
+        String updateVersionUrl = getVersionServiceUrl();
+        try {
+            URL url = new URL(updateVersionUrl);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            connection.setRequestProperty("User-Agent", "Kaazing Update Property");
+            connection.setRequestProperty("Connection", "close");
+            connection.setRequestProperty("Accept", "text/plain");
+            int responseCode = connection.getResponseCode();
+            if (responseCode >= 200 && responseCode <= 300) {
+                try (BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), "UTF-8"))) {
+                    JSONTokener tokener = new JSONTokener(br);
+                    JSONObject root = new JSONObject(tokener);
+                    String version = root.getString("version");
+                    latestVersion = parseGatewayVersion(version);
+                }
+            } else {
+                throw new Exception(format("Unexpected %d response code from versioning property", responseCode));
```

Comment:
Never throw Exception (this should be removed in the end as a result of the reformatting described on line 79